src/vi/vlogtruyen/src/eu/kanade/tachiyomi/extension/vi/vlogtruyen/VlogTruyen.kt

```diff
@@ -0,0 +1,228 @@
+    }
+
+    override fun latestUpdatesNextPageSelector() = ".pagination > li.active + li"
+
+    override fun latestUpdatesSelector() = "div.content-tab ul li.commic-hover"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/the-loai/dang-hot?page=$page")
+    }
+
+    override fun popularMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+
+    override fun popularMangaNextPageSelector() = latestUpdatesNextPageSelector()
+
+    override fun popularMangaSelector() = latestUpdatesSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.select("h1.title-commic-detail").text()
+        genre = document.select(".categories-list-detail-commic > li > a").joinToString { it.text().trim(',', ' ') }
+        description = document.select("div.top-detail-manga > div.top-detail-manga-content > span.desc-commic-detail").text()
+        thumbnail_url = document.select("div.image-commic-detail > a > img").attr("data-src")
+        status = parseStatus(document.select("div.top-detail-manga > div.top-detail-manga-avatar > div.manga-status > p").firstOrNull()?.text())
+        author = document.select(".h5-drawer:contains(Tác Giả) + ul li a").joinToString { it.text() }
+    }
+
+    private fun parseStatus(status: String?) = when {
+        status == null -> SManga.UNKNOWN
+        status.contains("Đang tiến hành") -> SManga.ONGOING
+        status.contains("Đã hoàn thành") -> SManga.COMPLETED
+        status.contains("Tạm ngưng") -> SManga.ON_HIATUS
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val html = response.body.string().replace("\\", "").replace("n  ", "").substringAfter("\"chaptersHtml\":\"").substringBefore("\",")
+        val document = response.asJsoup(html)
+        val hidePaidChapters = preferences.getBoolean(KEY_HIDE_PAID_CHAPTERS, false)
+        return document.select("li, .ul-list-chaper-detail-commic li").filterNot {
+            if (hidePaidChapters) {
+                it.select("li > b").text().isNotBlank().or(!hidePaidChapters)
+            } else {
+                it.select("li > a").text().isBlank().or(false)
+            }
+        }
+            .mapNotNull {
+                SChapter.create().apply {
+                    setUrlWithoutDomain(it.select("a").first()!!.attr("href"))
+                    name = it.select("h3").first()!!.text().replaceAfter("-", "").replaceAfter(":", "")
+                        .replaceAfter("0 ", "").replaceAfter("1 ", "").replaceAfter("2 ", "")
+                        .replaceAfter("3 ", "").replaceAfter("4 ", "").replaceAfter("5 ", "")
+                        .replaceAfter("6 ", "").replaceAfter("7 ", "").replaceAfter("8 ", "")
+                        .replaceAfter("9 ", "").replace("-", "").replace(":", "")
```

Comment:
why?

Reply:
I only show the name of the chapter. I just clean text ~~
![image](https://github.com/user-attachments/assets/d6257051-1592-4910-899f-5794cdf104d0)

Reply:
consider using regex